integration-cli/docker_cli_set_unix_test.go

```diff
@@ -0,0 +1,121 @@
+	}
+
+	file := "/sys/fs/cgroup/memory/memory.limit_in_bytes"
+	out, _ := dockerCmd(c, "exec", name, "cat", file)
+	c.Assert(strings.TrimSpace(out), checker.Equals, "524288000")
+}
+
+func (s *DockerSuite) TestSetRunningContainerWithRestart(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	testRequires(c, memoryLimitSupport)
+
+	name := "test-set-container"
+	dockerCmd(c, "run", "-d", "--name", name, "-m", "300M", "busybox", "top")
+	dockerCmd(c, "set", "-m", "500M", name)
+	dockerCmd(c, "restart", name)
+
+	memory, err := inspectField(name, "HostConfig.Memory")
+	c.Assert(err, check.IsNil)
+	if memory != "524288000" {
+		c.Fatalf("Got the wrong memory value, we got %d, expected 524288000(500M).", memory)
+	}
+
+	file := "/sys/fs/cgroup/memory/memory.limit_in_bytes"
+	out, _ := dockerCmd(c, "exec", name, "cat", file)
+	c.Assert(strings.TrimSpace(out), checker.Equals, "524288000")
+}
+
+func (s *DockerSuite) TestSetStoppedContainer(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	testRequires(c, memoryLimitSupport)
+
+	name := "test-set-container"
+	file := "/sys/fs/cgroup/memory/memory.limit_in_bytes"
+	dockerCmd(c, "run", "-d", "--name", name, "-m", "300M", "busybox", "cat", file)
+	dockerCmd(c, "wait", name)
+	dockerCmd(c, "set", "-m", "500M", name)
+
+	memory, err := inspectField(name, "HostConfig.Memory")
+	c.Assert(err, check.IsNil)
+	if memory != "524288000" {
+		c.Fatalf("Got the wrong memory value, we got %d, expected 524288000(500M).", memory)
+	}
+
+	out, _ := dockerCmd(c, "start", "-a", name)
+	c.Assert(strings.TrimSpace(out), checker.Equals, "524288000")
+}
+
+func (s *DockerSuite) TestSetPausedContainer(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	testRequires(c, cpuShare)
+
+	name := "test-set-container"
+	dockerCmd(c, "run", "-d", "--name", name, "--cpu-shares", "1000", "busybox", "top")
+	dockerCmd(c, "pause", name)
+	dockerCmd(c, "set", "--cpu-shares", "500", name)
+
+	out, err := inspectField(name, "HostConfig.CPUShares")
+	c.Assert(err, check.IsNil)
+	if out != "500" {
+		c.Fatalf("Got the wrong cpu shares value, we got %d, expected 500.", out)
+	}
+
+	dockerCmd(c, "unpause", name)
+	file := "/sys/fs/cgroup/cpu/cpu.shares"
+	out, _ = dockerCmd(c, "exec", name, "cat", file)
+	c.Assert(strings.TrimSpace(out), checker.Equals, "500")
+}
+
+func (s *DockerSuite) TestSetWithUntouchedFields(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+	testRequires(c, memoryLimitSupport)
+	testRequires(c, cgroupCpuset)
+
+	name := "test-set-container"
+	dockerCmd(c, "run", "-d", "--name", name, "-m", "300M", "--cpuset-cpus", "0", "busybox", "top")
+	dockerCmd(c, "set", "-m", "500M", name)
+
+	// Set memory and not touch cpus, `cpuset.cpus` should still have the old value
+	out, err := inspectField(name, "HostConfig.CpusetCpus")
+	c.Assert(err, check.IsNil)
+	c.Assert(out, check.Equals, "0")
+
+	file := "/sys/fs/cgroup/cpuset/cpuset.cpus"
+	out, _ = dockerCmd(c, "exec", name, "cat", file)
+	c.Assert(strings.TrimSpace(out), checker.Equals, "0")
```

Comment:
I think this test would be better if cpuset-cpus was set to "1" on line 98 and then we checked to make sure it was still "1" at this point.